src/discussions/posts/data/thunks.js

```diff
@@ -67,7 +106,8 @@ export function fetchThreads(courseId, {
     try {
       dispatch(fetchThreadsRequest({ courseId }));
       const data = await getThreads(courseId, options);
-      dispatch(fetchThreadsSuccess(camelCaseObject(data)));
+      const normalisedData = normaliseThreads(camelCaseObject(data));
```

Comment:
Everywhere in the code base, we are normalizing data in API actions. what do you think is the best place to normalize data in thunks or in API actions?

Reply:
I think thunks are a better place since it's possible to have two separate thunks that use the same kind of API differently and need different kinds of normalisation. So I try to keep the API side very raw and make it just a mapping of an API to function. The thunks then convert that into an operation on state. So I can imagine a thunk doing multiple API calls for instance, if needed and normalising them together.

Reply:
Agreed. 👍